### PR TITLE
Improvement/deprecate inject metadata

### PR DESCRIPTION
**Pull request by pacmessica**
[notion](https://www.notion.so/figmentnetworks/b47c0f8b935741d8ac28717ddeea2038?v=5e4e1d3854094b788ae2b612329f27e8&p=14d4b430a71643deaea3bfde9ed3aa0e)

### CODE REVIEW

**Review by silentlight — APPROVED** [review]
This is awesome @pacmessica! I am so glad we don't have to deal with this metadata anymore. Also, thanks for performance optimization :)

File: handlers/chain_handlers.js [hunk 1]
```diff
@@ -64,34 +62,26 @@ const getStatus = async (api, _call, context = {}) => {
...
+  const blockHash = context.blockHash ? context.blockHash : await getHashForHeight(api, height);
+
+  const [chain, runtimeVersionAt, rawCurrentEra, rawCurrentSession, rawNextEra, rawNextSession, rawTimestampAt
+  ] = await Promise.all([
+    api.rpc.system.chain(),
+    api.rpc.state.getRuntimeVersion(prevBlockHash),
+    // To get current session and era we need to use previous block hash
+    api.query.staking.currentEra.at(prevBlockHash),
+    api.query.session.currentIndex.at(prevBlockHash),
+  // Current height block hash gets next era and session
```
Comment (pawelnguyen) [inline handlers/chain_handlers.js:75]: indentation ;)

File: utils/block.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+const getHashForHeight = async(api,height) => {
+    return (!height || height === 0)
+        ? await api.rpc.chain.getFinalizedHead()
+        : await api.rpc.chain.getBlockHash(height);
+}
+
+module.exports = {
+getHashForHeight
+};
+
```
Comment (pawelnguyen) [inline utils/block.js:11]: here also indentation is off

**Review by pawelnguyen — COMMENTED** [review]
nice, things should be faster. If I remember correctly injecting metadata was needed because of querying for older data. Is this solved now by polkadotjs?

also have you tried running tests? I think there's some basic ones, but since CI isn't set up they might be out of date?

**Comment by pacmessica** [comment]
> nice, things should be faster. If I remember correctly injecting metadata was needed because of querying for older data. Is this solved now by polkadotjs?
> 
> also have you tried running tests? I think there's some basic ones, but since CI isn't set up they might be out of date?

correct, polkadotjs now handles it.

I think the tests aren't up to date here, I'll give it a shot though. Some calls only work for a certain history depth (~80 days), and then after that they return 0. So the snapshots in the tests we have won't last unfortunately.

I tested all the calls locally by comparing before/after data using historical calls and only caught one bug in the handlers/validators. Also been using these changes for the past week while developing rewards and haven't had any issues.

File: utils/block.js (hunk 2 above)
Comment (pacmessica) [reply utils/block.js:11]: so used to linters doing this for me 😅 . Thanks!

**Pull request merged**